packages/notification-services-controller/src/NotificationServicesController/NotificationServicesController.ts

```diff
@@ -1096,35 +1096,46 @@ export default class NotificationServicesController extends BaseController<
     try {
       this.#setIsFetchingMetamaskNotifications(true);
 
+      // This is used by Feature Announcement & On Chain
+      // Not used by Snaps
+      const isGlobalNotifsEnabled = this.state.isNotificationServicesEnabled;
+
       // Raw Feature Notifications
-      const rawFeatureAnnouncementNotifications = this.state
-        .isFeatureAnnouncementsEnabled
-        ? await FeatureNotifications.getFeatureAnnouncementNotifications(
-            this.#featureAnnouncementEnv,
-            previewToken,
-          ).catch(() => [])
-        : [];
+      const rawFeatureAnnouncementNotifications =
+        isGlobalNotifsEnabled && this.state.isFeatureAnnouncementsEnabled
+          ? await FeatureNotifications.getFeatureAnnouncementNotifications(
+              this.#featureAnnouncementEnv,
+              previewToken,
+            ).catch(() => [])
+          : [];
 
       // Raw On Chain Notifications
       const rawOnChainNotifications: OnChainRawNotification[] = [];
-      const userStorage = await this.#storage
-        .getNotificationStorage()
-        .then((s) => s && (JSON.parse(s) as UserStorage))
-        .catch(() => null);
-      const bearerToken = await this.#auth.getBearerToken().catch(() => null);
-      if (userStorage && bearerToken) {
-        const notifications =
-          await OnChainNotifications.getOnChainNotifications(
-            userStorage,
-            bearerToken,
-          ).catch(() => []);
-
-        rawOnChainNotifications.push(...notifications);
+      if (isGlobalNotifsEnabled) {
```

Comment:
Surrounded code with this `isGlobalNotifsEnabled` check to ensure we fetch wallet notifications if the global flag is enabled.